ocollection.h

```diff
@@ -0,0 +1,90 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+struct OVector{
+    int size;
+    int capacity;
+    int* array;
+};
+
+void initializeVector(struct OVector* vector) {
+    vector->size = 0;
+    vector->capacity = 5;
+    vector->array = malloc(vector->capacity * sizeof(int));
+}
+
+int* createArray( int capacity) {
+    int* array = malloc(capacity+sizeof(int));
+    return array;
+}
+
+void deleteArray(struct OVector* vector) {
+    free(vector->array);
+}
+
+void resizeArray(struct OVector* vector, int newCapacity) {
+    int* newArray = createArray(newCapacity);
+    for (int i = 0; i < vector->capacity; i++) {
+        newArray[i] = vector->array[i];
+    }
+    deleteArray(vector);
+    vector->array = newArray;
+    vector->capacity = newCapacity;
+}
+
+void push_back(struct OVector* vector, int element) {
+    if (vector->size == vector->capacity) {
```

Comment:
you using this type of check in many places. Maybe will be good to mov it to some function?